### ISSUE TITLE
~/.helm does not follow XDG standards

### ISSUE THREAD

**Issue by hereforthebugs** [helm/helm#4491]
As it stands now, helm makes a .helm folder in the home directory. 
```
.helm/
├── Helm.config
└── patches

1 directory, 1 file
```

[XDG standards](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) specify that config files should go in ~/.config and data in ~/.local/share. 

The entirety of ~/.helm should be moved to ~/.local/share/helm, because it contains application data and Helm.config really doesn't seem to be much of a config file, containing only a version number and data used to determine if the user should be asked for payment.

Great work guys, keep it up.

**Comment by bacongobbler** [comment]
Hey @hereforthebugs! Great handle btw ;)

I haven't heard of Helm.config before. Perhaps you're thinking of [emacs-helm](https://github.com/emacs-helm/helm)?

Either way, we also store config information in ~/.helm so this is good feedback regardless.

**Comment by hereforthebugs** [comment]
@bacongobbler 
Come to think of it that might have been generated when I loaded helm into ardour or something.
Here are it's contents though
```
{
  "synth_version": "0.9.0",
  "day_asked_for_payment": 17737,
  "should_ask_for_payment": false
}
```

**Comment by technosophos** [comment]
So on two disconnected topics:

1. That's definitely not a config file generated by this project
2. We don't follow the XDG spec, though we could in Helm 3. We need to make sure, though, that such a solution is acceptable to the Windows crowd (can't imagine it would not be)

If we followed XDG, we'd have to put repository caches in `.local/share/helm` and put config in `.config/helm`. That seems reasonable.

/cc @adamreese

**Comment by adamreese** [comment]
Yay!  I was going to sneak this into Helm3...

XDG_CACHE_HOME locations are different based on OS

```
unixish: $HOME/.cache
darwin:  $XDG_CACHE_HOME or $HOME/Library/Caches
windows: %LocalAppData%
```

**Comment by mattfarina** [comment]
Anyone want to take this on?

**Comment by cleac** [comment]
I'd like to take this, if possible :wink:

**Comment by bacongobbler** [comment]
IIRC @adamreese may already be working on this, so I'd reach out and make sure it already hasn't been implemented for Helm 3

**Comment by adamreese** [comment]
I am not working on this currently. Go for it @cleac

**Comment by cleac** [comment]
Working on this. To make the PR into v3, I should use dev-v3 branch, right?

**Comment by cleac** [comment]
I've made small proposal of changes, I'm gonna test it further a bit later: https://github.com/helm/helm/pull/5166

Feel free to tell, what should be improved, due to I'm not pro in Go :wink:

**Comment by bacongobbler** [comment]
closed via #5443
